extractor/filesystem/language/python/poetrylock/poetrylock.go

```diff
@@ -65,6 +66,28 @@ func (e Extractor) FileRequired(api filesystem.FileAPI) bool {
 	return filepath.Base(api.Path()) == "poetry.lock"
 }
 
+func resolveGroups(pkg poetryLockPackage) []string {
+	// by definition an optional package cannot be in any other group,
```

Comment:
I'm not clear on how optional works here, does this refer to this? https://python-poetry.org/docs/managing-dependencies/#optional-groups or are optional groups and optional packages different.

Reply:
umm ok I didn't know about that - I believe it was referring to https://python-poetry.org/docs/pyproject/#optional-dependencies but I'll check what happens when using these optional groups.

fwiw right now "optional" logically is focused on "is the `optional` property set to `true`" - so I know that `optional-dependencies` is one way of having that property come out as `true`, and now the question is if "optional groups" result in that too or something different

Reply:
it looks like "optional groups" just manage what gets installed by default when you run `poetry install`, so they don't have any bearing on the lockfile itself, which makes sense